grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -500,35 +500,60 @@ class StockMovementService {
 
     void removeStockMovementItem(String id) {
         RequisitionItem requisitionItem = RequisitionItem.get(id)
-        removeRequisitionItem(requisitionItem)
+        ShipmentItem shipmentItem = ShipmentItem.get(id)
+        if (requisitionItem) {
+            removeRequisitionItem(requisitionItem)
+        } else {
+            removeShipmentItem(shipmentItem)
+        }
     }
 
     def getStockMovementItems(String id, String stepNumber, String max, String offset) {
         // FIXME should get stock movement instead of requisition
+        Shipment shipment = Shipment.get(id)
         Requisition requisition = Requisition.get(id)
         List<StockMovementItem> stockMovementItems = []
         List <RequisitionItem> requisitionItems = []
+        List <ShipmentItem> shipmentItems = []
 
         if (stepNumber == '3') {
           return getEditPageItems(requisition, max, offset)
         }
 
-        if (max != null && offset != null) {
-            requisitionItems = RequisitionItem.createCriteria().list(max: max.toInteger(), offset: offset.toInteger()) {
-                eq("requisition", requisition)
-                isNull("parentRequisitionItem")
-                order("orderIndex", 'asc')
+        if (requisition) {
+            if (max != null && offset != null) {
+                requisitionItems = RequisitionItem.createCriteria().list(max: max.toInteger(), offset: offset.toInteger()) {
+                    eq("requisition", requisition)
+                    isNull("parentRequisitionItem")
+                    order("orderIndex", 'asc')
+                }
+            } else {
+                requisitionItems = RequisitionItem.createCriteria().list() {
+                    eq("requisition", requisition)
+                    isNull("parentRequisitionItem")
+                    order("orderIndex", 'asc')
+                }
+            }
+            requisitionItems.each { requisitionItem ->
+                StockMovementItem stockMovementItem = StockMovementItem.createFromRequisitionItem(requisitionItem)
+                stockMovementItems.add(stockMovementItem)
             }
         } else {
-            requisitionItems = RequisitionItem.createCriteria().list() {
-                eq("requisition", requisition)
-                isNull("parentRequisitionItem")
-                order("orderIndex", 'asc')
+            if (max != null && offset != null) {
```

Comment:
Why isn't this just 
```
if (max && offset) { 
```

Reply:
If there's a reason (offset can be 0 and therefore we need to be explicit) then I'm ok with it but I really don't like null comparisons after developing with Groovy for 10 years.

Reply:
Just as you said, offset can be 0 and that's why we're checking if it's not null.